src/java_bytecode/java_string_library_preprocess.cpp

```diff
@@ -30,24 +30,26 @@ Date:   April 2017
 
 #include "java_string_library_preprocess.h"
 
+/// \return tag of a struct prefixed by "java::" or symbolic tag
+/// empty string if not symbol or struct
+irep_idt get_tag(const typet &type)
+{
+  if(type.id()==ID_symbol)
+    return to_symbol_type(type).get_identifier();
+  else if(type.id()==ID_struct)
+    return irep_idt("java::"+id2string(to_struct_type(type).get_tag()));
```

Comment:
Rather than assume this symbol <-> tag relationship, just ns.follow and return the real tag